src/account/accountReducers.js

```diff
@@ -74,7 +74,26 @@ export default (state: AccountState = initialState, action: Action): AccountStat
 
     case LOGOUT: {
       // Empty out the active account's api key
-      return [{ ...state[0], apiKey: '' }, ...state.slice(1)];
+      const { auth } = action;
+      let isStateChanged = false;
+      const newState = state.reduce((accounts, account) => {
+        if (
+          account.apiKey !== '' &&
+          account.email === auth.email &&
+          account.apiKey === auth.apiKey
+        ) {
+          isStateChanged = true;
+          accounts.push({ ...account, apiKey: '' });
+        } else {
+          accounts.push(account);
+        }
+        return accounts;
+      }, []);
+
+      if (isStateChanged) {
+        return newState;
+      }
+      return state;
```

Comment:
This is kind of a lot of code for what's ultimately doing something pretty simple.

How about [Array#map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map)?

The one thing a simple use of `map` wouldn't get is the `isStateChanged` flag. But is that helping us much?

Reply:
The original code is good as-is.
We are doing perf optimizations in several places similar to what @jainkuniya did here.
But it is in only very specific cases where we performance matters.
This is not one of the cases